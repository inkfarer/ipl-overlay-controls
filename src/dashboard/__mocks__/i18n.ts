import i18next from 'i18next';
import { config } from '@vue/test-utils';

jest.mock('i18next-vue', () => ({
    __esModule: true,
    useTranslation: () => ({ t: i18next.t })
}));

beforeAll(async () => {
    await i18next.init({
        lng: 'cimode',
        appendNamespaceToCIMode: true
    });
    config.global.mocks = {
        $t: i18next.t
    };
});
