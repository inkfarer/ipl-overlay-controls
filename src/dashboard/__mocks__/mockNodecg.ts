import Mock = jest.Mock;

export type ReplicantChangeHandler = (newValue?: unknown, oldValue?: unknown) => void;
export type MockReplicant = { value: unknown };
type OnFunction = (event: string, handler: ReplicantChangeHandler) => void;

export class MockNodecg {
    replicants: {[key: string]: MockReplicant};
    listeners: {[key: string]: ReplicantChangeHandler};
    sendMessage: Mock

    constructor() {
        this.replicants = {};
        this.listeners = {};
        this.sendMessage = jest.fn();
    }

    init(): void {
        const self = this;
        window.nodecg = {
            // @ts-ignore: Just a mock
            Replicant(name: string) {
                const replicantValue: { value: unknown, on: OnFunction } = {
                    value: undefined,
                    on: (event: string, handler: ReplicantChangeHandler) => {
                        self.listeners[name] = handler;
                    }
                };
                self.replicants[name] = replicantValue;
                return replicantValue;
            },
            sendMessage: self.sendMessage,
            // @ts-ignore: Just a mock
            log: {
                info: jest.fn()
            }
        };

        // @ts-ignore: It's a mock.
        window.NodeCG = {
            waitForReplicants: jest.fn().mockResolvedValue(true)
        };
    }
}
