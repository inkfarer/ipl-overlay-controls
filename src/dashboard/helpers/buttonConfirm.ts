/*
buttonConfirm

After importing this script to a dashboard panel, add the attribute 'data-requires-confirmation' to any button.
The button will now ask for confirmation when it is first clicked.
When it is clicked a second time, a 'confirm' event is sent by it, which can be listened for by scripts.

Example:

HTML:
    <button id="dangerous-action-button" data-requires-confirmation>DO A THING</button>
JS:
    document.getElementById('dangerous-action-button').addEventListener('confirm', () => {
        // do something
    });
*/

window.addEventListener('DOMContentLoaded', () => {
    function addListener(btn: HTMLButtonElement) {
        if (btn.dataset.hasConfirmListener !== 'true') {
            btn.addEventListener('click', e => {
                const target = e.target as HTMLButtonElement;

                function resetButton() {
                    target.dataset.isClicked = 'false';
                    target.innerText = target.dataset.originalText;
                    target.style.backgroundColor = '';
                }

                if (target.dataset.isClicked === 'true') {
                    target.dispatchEvent(new Event('confirm'));
                    resetButton();
                } else {
                    target.dataset.isClicked = 'true';
                    target.dataset.originalText = target.innerText;
                    target.style.backgroundColor = 'var(--red)';
                    target.innerText = 'are you sure?';

                    setTimeout(() => {
                        resetButton();
                    }, 5000);
                }
            });

            btn.dataset.hasConfirmListener = 'true';
        }
    }

    const buttons = document.querySelectorAll('button[data-requires-confirmation]');
    buttons.forEach(addListener);

    const observer = new MutationObserver(mutations => {
        mutations.forEach(mutation => {
            mutation.addedNodes.forEach(node => {
                if (node.nodeType === Node.ELEMENT_NODE
                    && node.nodeName === 'BUTTON'
                    && (node as HTMLElement).dataset.requiresConfirmation) {

                    const btn = node as HTMLButtonElement;
                    addListener(btn);
                }
            });
        });
    });

    observer.observe(document.body, {
        subtree: true,
        childList: true
    });
});
