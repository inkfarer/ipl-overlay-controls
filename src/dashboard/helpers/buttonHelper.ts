import { colors } from '../styles/colors';

export function setToggleButtonDisabled(
    toggleTrue: HTMLButtonElement,
    toggleFalse: HTMLButtonElement,
    state: boolean
): void {
    toggleTrue.disabled = state;
    toggleFalse.disabled = !state;
}

export function addChangeReminder(
    elements: HTMLElement[] | NodeListOf<HTMLElement>,
    updateButton: HTMLButtonElement
): void {
    elements.forEach(element => {
        if (!element.tagName) return;

        let event;
        if (element.tagName.toLowerCase() === 'input') event = 'input';
        else if (element.tagName.toLowerCase() === 'select') event = 'change';
        else return;

        element.addEventListener(event, e => {
            const target = e.target as HTMLElement;
            target.dataset.edited = 'true';
            updateButton.style.backgroundColor = colors.red;
        });
    });

    updateButton.addEventListener('click', () => {
        elements.forEach(elem => {
            elem.dataset.edited = 'false';
        });
        updateButton.style.backgroundColor = colors.blue;
    });
}
