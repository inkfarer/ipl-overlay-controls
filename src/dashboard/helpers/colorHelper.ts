export function getContrastingTextColor(backgroundColor: string, lightColor = 'white', darkColor = '#333'): string {
    const { r, g, b } = hexToRbg(backgroundColor);
    const yiq = ((r * 299) + (g * 587) + (b * 114)) / 1000;
    return (yiq >= 128) ? darkColor : lightColor;
}

export function hexToRbg(color: string): { r: number, g: number, b: number } {
    color = color.replace('#', '');
    return {
        r: parseInt(color.substr(0, 2), 16),
        g: parseInt(color.substr(2, 2), 16),
        b: parseInt(color.substr(4, 2), 16)
    };
}
