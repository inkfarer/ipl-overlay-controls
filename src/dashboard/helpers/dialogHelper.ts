import { NodecgDialog } from '../types/dialog';

export function closeDialog(dialogName: string): void {
    (nodecg.getDialog(dialogName) as NodecgDialog).close();
}
