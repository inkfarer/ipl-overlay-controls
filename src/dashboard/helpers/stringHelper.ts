export function addDots(value: string, maxLength = 48): string {
    const rolloff = '...';

    if (!value) return value;
    if (value.length > maxLength) {
        return value.substring(0, maxLength - rolloff.length) + rolloff;
    }

    return value;
}
