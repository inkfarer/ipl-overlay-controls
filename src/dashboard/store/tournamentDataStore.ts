import { MatchStore, RoundStore, TournamentData } from 'schemas';
import {
    GetTournamentDataRequest,
    GetTournamentDataResponse,
    ToggleTeamImageRequest
} from 'types/messages/tournamentData';
import { UpdateRoundStoreRequest, UpdateRoundStoreResponse } from 'types/messages/roundStore';
import { defineStore } from 'pinia';
import i18next from 'i18next';

const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');
const roundStore = nodecg.Replicant<RoundStore>('roundStore');
const matchStore = nodecg.Replicant<MatchStore>('matchStore');

export const tournamentDataReps = [ tournamentData, roundStore, matchStore ];

export interface TournamentDataStore {
    tournamentData: TournamentData;
    roundStore: RoundStore;
    matchStore: MatchStore;
}

export const useTournamentDataStore = defineStore('tournamentData', {
    state: () => ({
        tournamentData: null,
        roundStore: null,
        matchStore: null
    } as TournamentDataStore),
    actions: {
        async getTournamentData({ method, id }: GetTournamentDataRequest): Promise<GetTournamentDataResponse> {
            return nodecg.sendMessage('getTournamentData', { method, id });
        },
        async getSmashggEvent({ eventId }: { eventId: number }): Promise<void> {
            return await nodecg.sendMessage('getSmashggEvent', { eventId });
        },
        async uploadTeamData({ file }: { file: File }): Promise<void> {
            return sendLocalFile('teams', file);
        },
        async uploadRoundData({ file }: { file: File }): Promise<void> {
            return sendLocalFile('rounds', file);
        },
        async fetchRoundData({ url }: { url: string }): Promise<void> {
            return nodecg.sendMessage('getRounds', { url });
        },
        async updateRound(data: UpdateRoundStoreRequest): Promise<void> {
            return nodecg.sendMessage('updateRound', data);
        },
        async insertRound(data: UpdateRoundStoreRequest): Promise<UpdateRoundStoreResponse> {
            return nodecg.sendMessage('insertRound', data);
        },
        resetRoundStore() {
            nodecg.sendMessage('resetRoundStore');
        },
        setTeamImageHidden({ teamId, isVisible }: { teamId: string, isVisible: boolean }): Promise<void> {
            return nodecg.sendMessage('toggleTeamImage', { teamId, isVisible } as ToggleTeamImageRequest);
        },
        removeRound({ roundId }: { roundId: string }): Promise<void> {
            return nodecg.sendMessage('removeRound', { roundId });
        },
        setShortName(newValue: string): void {
            tournamentData.value.meta.shortName = newValue;
        },
        async refreshTournamentData(): Promise<void> {
            return nodecg.sendMessage('refreshTournamentData');
        }
    }
});

async function sendLocalFile(dataType: 'teams' | 'rounds', file: File): Promise<void> {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('jsonType', dataType);

    const response = await fetch('/ipl-overlay-controls/upload-tournament-json', { method: 'POST', body: formData });

    if (response.status !== 200) {
        const responseText = await response.text();
        throw new Error(i18next.t(
            'common:tournamentDataStore.fileUploadError',
            { statusCode: response.status, message: responseText }));
    }
}
