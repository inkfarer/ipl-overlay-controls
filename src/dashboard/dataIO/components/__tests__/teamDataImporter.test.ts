import TeamDataImporter from '../teamDataImporter.vue';
import { TournamentDataSource } from 'types/enums/tournamentDataSource';
import { config, flushPromises, mount } from '@vue/test-utils';
import { mockBundleConfig } from '../../../__mocks__/mockNodecg';
import * as stringHelper from '../../../helpers/stringHelper';
import { createTestingPinia, TestingPinia } from '@pinia/testing';
import { useTournamentDataStore } from '../../../store/tournamentDataStore';

describe('teamDataImporter', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        IplButton: true,
        IplSelect: true,
        IplInput: true,
        IplUpload: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();

        useTournamentDataStore().$state = {
            tournamentData: {
                meta: {
                    id: '123123',
                    source: TournamentDataSource.BATTLEFY,
                    shortName: null,
                    name: 'Tournament Name'
                },
                teams: []
            },
            roundStore: {},
            matchStore: {}
        };
    });

    it('matches snapshot', () => {
        const store = useTournamentDataStore();
        store.tournamentData.meta = {
            source: TournamentDataSource.BATTLEFY,
            id: '123123asd',
            name: 'Cool Tournament',
            shortName: 'Cool Tournament',
            url: 'tournament://cool-tournament'
        };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('displays data of existing tournament', () => {
        const store = useTournamentDataStore();
        store.tournamentData.meta = {
            source: TournamentDataSource.BATTLEFY,
            id: '123123asd',
            name: 'Cool Tournament',
            shortName: 'Cool Tournament',
            url: 'tournament://cool-tournament'
        };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.get('.existing-data').html()).toMatchSnapshot();
    });

    it('displays data of existing tournament if it is missing a name', () => {
        const store = useTournamentDataStore();
        store.tournamentData.meta = {
            source: TournamentDataSource.BATTLEFY,
            id: '123123asd',
            url: 'tounament://cool-tournament',
            shortName: null
        };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.get('.existing-data').html()).toMatchSnapshot();
    });

    it('displays data of existing tournament if it is missing an URL', () => {
        const store = useTournamentDataStore();
        store.tournamentData.meta = {
            source: TournamentDataSource.BATTLEFY,
            id: '123123asd',
            name: 'cool tournament',
            shortName: 'cool tournament'
        };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.get('.existing-data').html()).toMatchSnapshot();
    });

    it('displays data of existing tournament if it has smash.gg data', () => {
        const store = useTournamentDataStore();
        store.tournamentData.meta = {
            source: TournamentDataSource.SMASHGG,
            id: '123123asd',
            name: 'cool tournament',
            shortName: 'cool tournament',
            url: 'smashgg://cool-tournament',
            sourceSpecificData: {
                smashgg: {
                    tournamentId: 1234,
                    eventData: {
                        id: 1234,
                        name: 'Cool Event',
                        game: 'Splatoon 2'
                    }
                }
            }
        };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.get('.existing-data').html()).toMatchSnapshot();
    });

    it('has expected data source options if smash.gg config is present', () => {
        mockBundleConfig.smashgg = { apiKey: 'ggkey123' };
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });
        const sourceSelector = wrapper.getComponent('[data-test="source-selector"]');

        expect((sourceSelector.vm.$props as { options: unknown }).options).toEqual([
            { name: 'Battlefy', value: TournamentDataSource.BATTLEFY },
            { name: 'Smash.gg', value: TournamentDataSource.SMASHGG },
            { name: 'Uploaded file', value: TournamentDataSource.UPLOAD }
        ]);
    });

    it('has expected data source options if smash.gg config is not present', () => {
        mockBundleConfig.smashgg = undefined;
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });
        const sourceSelector = wrapper.getComponent('[data-test="source-selector"]');

        expect((sourceSelector.vm.$props as { options: unknown }).options).toEqual([
            { name: 'Battlefy', value: TournamentDataSource.BATTLEFY },
            { name: 'Uploaded file', value: TournamentDataSource.UPLOAD }
        ]);
    });

    it('dispatches to store on import', () => {
        const store = useTournamentDataStore();
        store.getTournamentData = jest.fn();
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'SMASHGG');
        wrapper.getComponent('[name="tournament-id-input"]').vm.$emit('update:modelValue', 'cool-tourney');
        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');

        expect(store.getTournamentData).toHaveBeenCalledWith({
            method: TournamentDataSource.SMASHGG,
            id: 'cool-tourney'
        });
    });

    it('extracts battlefy tournament URL on import', () => {
        const store = useTournamentDataStore();
        store.getTournamentData = jest.fn();
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });
        jest.spyOn(stringHelper, 'extractBattlefyTournamentId').mockReturnValue('battlefy-tournament-id');

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'BATTLEFY');
        wrapper.getComponent('[name="tournament-id-input"]').vm.$emit('update:modelValue', 'cool-tourney');
        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');

        expect(store.getTournamentData).toHaveBeenCalledWith({
            method: TournamentDataSource.BATTLEFY,
            id: 'battlefy-tournament-id'
        });
        expect(stringHelper.extractBattlefyTournamentId).toHaveBeenCalled();
    });

    it('dispatches to store on import if file upload is enabled but source is not set to UPLOAD', async () => {
        const store = useTournamentDataStore();
        store.getTournamentData = jest.fn();
        store.uploadTeamData = jest.fn();
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'SMASHGG');
        wrapper.getComponent('[name="tournament-id-input"]').vm.$emit('update:modelValue', 'cool-tourney');
        wrapper.getComponent('[data-test="use-file-upload-checkbox"]').vm.$emit('update:modelValue', true);
        await wrapper.vm.$nextTick();
        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');

        expect(store.getTournamentData).toHaveBeenCalledWith({
            method: TournamentDataSource.SMASHGG,
            id: 'cool-tourney'
        });
        expect(store.uploadTeamData).not.toHaveBeenCalled();
    });

    it('dispatches to store on file import', async () => {
        const file = new File([], 'mock-file');
        const store = useTournamentDataStore();
        store.uploadTeamData = jest.fn();
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'UPLOAD');
        wrapper.getComponent('[data-test="use-file-upload-checkbox"]').vm.$emit('update:modelValue', true);
        wrapper.getComponent('[data-test="team-data-upload"]').vm.$emit('update:modelValue', file);
        await wrapper.vm.$nextTick();
        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');

        expect(store.uploadTeamData).toHaveBeenCalledWith({ file });
    });

    it('shows smash.gg event selector if import returns more than one event', async () => {
        const store = useTournamentDataStore();
        store.getTournamentData = jest.fn().mockResolvedValue({
            events: [
                { id: 123123, name: 'Event One', game: 'Splatoon 2' },
                { id: 456456, name: 'Event Two', game: 'Ultimate' }
            ]
        });
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');
        await flushPromises();

        expect(wrapper.findComponent('[data-test="smashgg-event-selector"]').exists()).toEqual(true);
        expect(wrapper.findComponent('[data-test="smashgg-event-import-button"]').exists()).toEqual(true);
        expect(wrapper.findComponent('[data-test="smashgg-event-import-cancel-button"]').exists()).toEqual(true);
        expect(wrapper.findComponent('[data-test="import-button"]').exists()).toEqual(false);
    });

    it('dispatches to store on smash.gg event import', async () => {
        const store = useTournamentDataStore();
        store.getSmashggEvent = jest.fn().mockResolvedValue({});
        store.getTournamentData = jest.fn().mockResolvedValue({
            events: [
                { id: 123123, name: 'Event One', game: 'Splatoon 2' },
                { id: 456456, name: 'Event Two', game: 'Ultimate' }
            ]
        });
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');
        await flushPromises();
        wrapper.getComponent('[data-test="smashgg-event-selector"]').vm.$emit('update:modelValue', '456456');
        wrapper.getComponent('[data-test="smashgg-event-import-button"]').vm.$emit('click');
        await flushPromises();

        expect(store.getSmashggEvent).toHaveBeenCalledWith({ eventId: 456456 });
        expect(wrapper.findComponent('[data-test="import-button"]').exists()).toEqual(true);
    });

    it('handles smash.gg event import being cancelled', async () => {
        const store = useTournamentDataStore();
        store.getSmashggEvent = jest.fn();
        store.getTournamentData = jest.fn().mockResolvedValue({
            events: [
                { id: 123123, name: 'Event One', game: 'Splatoon 2' },
                { id: 456456, name: 'Event Two', game: 'Ultimate' }
            ]
        });
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="import-button"]').vm.$emit('click');
        await flushPromises();
        wrapper.getComponent('[data-test="smashgg-event-import-cancel-button"]').vm.$emit('click');
        await wrapper.vm.$nextTick();

        expect(store.getSmashggEvent).not.toHaveBeenCalled();
        expect(wrapper.findComponent('[data-test="import-button"]').exists()).toEqual(true);
    });

    it('shows uploader if selected source is UPLOAD and file upload is enabled', async () => {
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'UPLOAD');
        wrapper.getComponent('[data-test="use-file-upload-checkbox"]').vm.$emit('update:modelValue', true);
        await wrapper.vm.$nextTick();

        expect(wrapper.findComponent('[data-test="team-data-upload"]').isVisible()).toEqual(true);
        expect(wrapper.findComponent('[name="tournament-id-input"]').isVisible()).toEqual(false);
    });

    it('shows ID input if selected source is UPLOAD and file upload is disabled', async () => {
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'UPLOAD');
        wrapper.getComponent('[data-test="use-file-upload-checkbox"]').vm.$emit('update:modelValue', false);
        await wrapper.vm.$nextTick();

        expect(wrapper.findComponent('[data-test="team-data-upload"]').isVisible()).toEqual(false);
        expect(wrapper.findComponent('[name="tournament-id-input"]').isVisible()).toEqual(true);
    });

    it('shows ID input if selected source is not UPLOAD and file upload is enabled', async () => {
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[data-test="use-file-upload-checkbox"]').vm.$emit('update:modelValue', true);
        wrapper.getComponent('[data-test="source-selector"]').vm.$emit('update:modelValue', 'BATTLEFY');
        await wrapper.vm.$nextTick();

        expect(wrapper.findComponent('[data-test="team-data-upload"]').isVisible()).toEqual(false);
        expect(wrapper.findComponent('[name="tournament-id-input"]').isVisible()).toEqual(true);
    });

    it('handles updating short name', async () => {
        const store = useTournamentDataStore();
        store.setShortName = jest.fn();
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });

        wrapper.getComponent('[name="shortName"]').vm.$emit('update:modelValue', 'Tournament Name');
        wrapper.getComponent('[data-test="update-short-name-button"]').vm.$emit('click');
        await wrapper.vm.$nextTick();

        expect(store.setShortName).toHaveBeenCalledWith('Tournament Name');
    });

    it('reverts short name changes on update button right click', async () => {
        const wrapper = mount(TeamDataImporter, {
            global: {
                plugins: [pinia]
            }
        });
        const event = new Event(null);
        jest.spyOn(event, 'preventDefault');

        wrapper.getComponent('[name="shortName"]').vm.$emit('update:modelValue', 'Tourney Name');
        wrapper.getComponent('[data-test="update-short-name-button"]').vm.$emit('right-click', event);
        await wrapper.vm.$nextTick();

        expect(wrapper.getComponent('[name="shortName"]').attributes().modelvalue).toEqual('Tournament Name');
        expect(event.preventDefault).toHaveBeenCalled();
    });
});
