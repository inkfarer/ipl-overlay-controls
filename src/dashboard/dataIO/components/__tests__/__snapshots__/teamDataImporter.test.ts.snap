// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teamDataImporter displays data of existing tournament if it has smash.gg data 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><a href="smashgg://cool-tournament" target="_blank">cool tournament</a><span class="badge badge-blue m-l-6">Smash.gg</span><br> 123123asd <br> Cool Event (Splatoon 2) <ipl-button-stub label="Refresh" color="blue" disabled="false" async="true" progressmessage="Refreshing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-6" data-test="refresh-button"></ipl-button-stub>
</div>"
`;

exports[`teamDataImporter displays data of existing tournament if it is missing a name 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><a href="tounament://cool-tournament" target="_blank">No name</a><span class="badge badge-blue m-l-6">Battlefy</span><br> 123123asd
  <!--v-if-->
  <ipl-button-stub label="Refresh" color="blue" disabled="false" async="true" progressmessage="Refreshing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-6" data-test="refresh-button"></ipl-button-stub>
</div>"
`;

exports[`teamDataImporter displays data of existing tournament if it is missing an URL 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><span>cool tournament</span><span class="badge badge-blue m-l-6">Battlefy</span><br> 123123asd
  <!--v-if-->
  <ipl-button-stub label="Refresh" color="blue" disabled="false" async="true" progressmessage="Refreshing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-6" data-test="refresh-button"></ipl-button-stub>
</div>"
`;

exports[`teamDataImporter displays data of existing tournament when its source is BATTLEFY 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><a href="tournament://cool-tournament" target="_blank">Cool Tournament</a><span class="badge badge-blue m-l-6">Battlefy</span><br> 123123asd
  <!--v-if-->
  <ipl-button-stub label="Refresh" color="blue" disabled="false" async="true" progressmessage="Refreshing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-6" data-test="refresh-button"></ipl-button-stub>
</div>"
`;

exports[`teamDataImporter displays data of existing tournament when its source is UNKNOWN 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><a href="tournament://cool-tournament" target="_blank">Cool Tournament</a><span class="badge badge-blue m-l-6">Unknown</span><br> 123123asd
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`teamDataImporter displays data of existing tournament when its source is UPLOAD 1`] = `
"<div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
  <div class="title">Saved data</div><a href="tournament://cool-tournament" target="_blank">Cool Tournament</a><span class="badge badge-blue m-l-6">Uploaded file</span><br> 123123asd
  <!--v-if-->
  <!--v-if-->
</div>"
`;

exports[`teamDataImporter matches snapshot 1`] = `
"<fieldset class="section-separator-wrapper">
  <legend>Tournament data</legend>
  <div data-v-11cc5f1c="" class="ipl-space color-dark">
    <ipl-radio-stub modelvalue="BATTLEFY" label="Source" options="[object Object],[object Object],[object Object]" data-test="source-selector" class="data-source-selector"></ipl-radio-stub>
    <ipl-input-stub name="tournament-id-input" label="Tournament URL" modelvalue="" type="text" centered="false" disabled="false" loading="false" class="m-t-4"></ipl-input-stub>
    <ipl-upload-stub class="m-t-8" data-test="team-data-upload" style="display: none;"></ipl-upload-stub>
    <div class="layout horizontal center-horizontal"><label data-v-15d39ed1="" class="small m-t-8" data-test="use-file-upload-checkbox" style="display: none;"><input data-v-15d39ed1="" type="checkbox"> Upload file</label></div>
    <ipl-button-stub label="Import" color="blue" disabled="true" async="true" progressmessage="Importing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="import-button"></ipl-button-stub>
  </div>
  <div data-v-11cc5f1c="" class="ipl-space color-dark text-center m-t-8" data-test="saved-data-section">
    <div class="title">Saved data</div><a href="tournament://cool-tournament" target="_blank">Cool Tournament</a><span class="badge badge-blue m-l-6">Battlefy</span><br> 123123asd
    <!--v-if-->
    <ipl-button-stub label="Refresh" color="blue" disabled="false" async="true" progressmessage="Refreshing..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-6" data-test="refresh-button"></ipl-button-stub>
  </div>
  <div data-v-11cc5f1c="" class="ipl-space color-dark m-t-8">
    <ipl-input-stub name="shortName" label="Short tournament name" modelvalue="Cool Tournament" type="text" centered="false" disabled="false" loading="false"></ipl-input-stub>
    <ipl-button-stub label="Update" color="blue" disabled="true" async="false" progressmessage="Working..." successmessage="Done!" requiresconfirmation="false" shortconfirmationmessage="false" small="false" disableonsuccess="false" href="" class="m-t-8" data-test="update-short-name-button" title="Right-click to undo changes."></ipl-button-stub>
  </div>
</fieldset>"
`;
